commit 2cb72f14b00b524f6d404bc5793d3378f888a521
Author: Sergei Trofimovich <slyfox@gentoo.org>
Date:   Mon Mar 3 16:46:33 2014 +0300

    src/Idris/ParseHelpers.hs: fix against ghc-7.8

    [50 of 75] Compiling Idris.ParseHelpers ( src/Idris/ParseHelpers.hs, dist/build/Idris/ParseHelpers.o )
    src/Idris/ParseHelpers.hs:39:97:
        Could not coerce from `Monad Parser' to `Monad IdrisInnerParser'
          because the first type argument of `Monad' has role Nominal,
          but the arguments `Parser' and `IdrisInnerParser' differ
          arising from the coercion of the method `notFollowedBy' from type
                       `forall a. (Monad Parser, Show a) => Parser a -> Parser ()' to type
                       `forall a.
                        (Monad IdrisInnerParser, Show a) =>
                        IdrisInnerParser a -> IdrisInnerParser ()'
        Possible fix:
          use a standalone 'deriving instance' declaration,
            so you can specify the instance context yourself
        When deriving the instance for (Parsing IdrisInnerParser)

    Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

diff --git a/src/Idris/ParseHelpers.hs b/src/Idris/ParseHelpers.hs
index 5cd3af8..567b32f 100644
--- a/src/Idris/ParseHelpers.hs
+++ b/src/Idris/ParseHelpers.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving, ConstraintKinds, PatternGuards #-}
+{-# LANGUAGE StandaloneDeriving #-}
 module Idris.ParseHelpers where
 
 import Prelude hiding (pi)
@@ -37,7 +38,18 @@ import System.FilePath
 type IdrisParser = StateT IState IdrisInnerParser
 
 newtype IdrisInnerParser a = IdrisInnerParser { runInnerParser :: Parser a }
-  deriving (Monad, Functor, MonadPlus, Applicative, Alternative, CharParsing, LookAheadParsing, Parsing, DeltaParsing, MarkParsing Delta, Monoid)
+
+deriving instance Monad IdrisInnerParser
+deriving instance Functor IdrisInnerParser
+deriving instance MonadPlus IdrisInnerParser
+deriving instance Applicative IdrisInnerParser
+deriving instance Alternative IdrisInnerParser
+deriving instance CharParsing IdrisInnerParser
+deriving instance LookAheadParsing IdrisInnerParser
+deriving instance Parsing IdrisInnerParser
+deriving instance DeltaParsing IdrisInnerParser
+deriving instance (MarkParsing Delta) IdrisInnerParser
+deriving instance Monoid a => Monoid (IdrisInnerParser a)
 
 instance TokenParsing IdrisInnerParser where
   someSpace = many (simpleWhiteSpace <|> singleLineComment <|> multiLineComment) *> pure ()
